### PR TITLE
Fixes incorrect implementation of DispatchTime.init(uptimeNanoseconds:)

### PR DESCRIPTION
**Pull request by ktopley-apple**
Internally, DispatchTime works in Mach absolute time, but the DispatchTime(uptimeNanoseconds:) initializer allows time to be given as nanoseconds since boot. This needs to be converted to Mach absolute time, but the conversion was not being performed. As a result, on platforms where the conversion factor is not 1, the wrong time is stored.

The fix is to convert the given time to Mach absolute time when the conversion factor is not 1. This may cause some loss of precision, so the uptimeNanoseconds property will not hold the same value as the one passed to the initializer, unless that value was an exact multiple of the conversion factor. One way to fix this would be to store both the actual value that was given and the converted value. However, dispatch does not use the original value and keeping it would add another field to the DispatchTime object, just on the off-chance that the application might want the exact initial value. Instead of saving the actual value, I added a comment to the initializer to indicate that the value read from the uptimeNanoseconds property will be the effective value, which might be rounded up from the original.

(Radar 29660448)
rdar://problem/29660448

### CODE REVIEW

**Comment by ktopley-apple** [comment]
@moiseev @mwwa @das please review

**Comment by ktopley-apple** [comment]
@swift-ci please smoke test

File: stdlib/public/SDK/Dispatch/Time.swift [hunk 1]
```diff
@@ -41,12 +42,22 @@ public struct DispatchTime : Comparable {
...
+	///               of the rounded result can be obtained by reading the
+	///               `uptimeNanoseconds` property.
 	///               Note that `DispatchTime(uptimeNanoseconds: 0)` is
 	///               equivalent to `DispatchTime.now()`, that is, its value
 	///               represents the number of nanoseconds since boot (excluding
 	///               system sleep time), not zero nanoseconds since boot.
 	public init(uptimeNanoseconds: UInt64) {
-		self.rawValue = dispatch_time_t(uptimeNanoseconds)
+		var rawValue = uptimeNanoseconds
+		if (DispatchTime.timebaseInfo.numer != DispatchTime.timebaseInfo.denom) {
```
Comment (CodaFi) [inline stdlib/public/SDK/Dispatch/Time.swift]: More of a stylistic thing, but at the expense of an `else` block you can make this immutable.

**Review by mwwa — APPROVED** [review]

**Comment by ktopley-apple** [comment]
@swift-ci please smoke test and merge

**Pull request merged**